src/value/owned.rs

```diff
@@ -5,159 +5,71 @@
 //! This API is simpler to manage with respect to borrowing lifetimes, but requires full
 //! ownership of data to do so.
 
-use super::{ImportSource, IonElement, IonSequence, IonStruct, IonSymbolToken};
+use super::{IonElement, IonSequence, IonStruct, IonSymbolToken};
 use crate::ion_eq::IonEq;
 use crate::text::text_formatter::IonValueFormatter;
 use crate::types::decimal::Decimal;
 use crate::types::integer::Integer;
 use crate::types::timestamp::Timestamp;
 use crate::types::SymbolId;
 use crate::value::Builder;
-use crate::IonType;
+use crate::{IonType, Symbol};
 use num_bigint::BigInt;
 use std::collections::HashMap;
 use std::fmt::{Display, Formatter};
 use std::iter::FromIterator;
 use std::rc::Rc;
 
-/// An owned implementation of  [`ImportSource`].
-#[derive(Debug, Clone)]
-pub struct OwnedImportSource {
-    table: Rc<str>,
-    sid: SymbolId,
-}
+impl IonSymbolToken for Symbol {
+    fn text(&self) -> Option<&str> {
+        self.text()
+    }
 
-impl OwnedImportSource {
-    pub fn new<T: Into<Rc<str>>>(table: T, sid: SymbolId) -> Self {
-        Self {
-            table: table.into(),
-            sid,
+    fn symbol_id(&self) -> Option<SymbolId> {
+        match self.text() {
+            Some(_) => None,
+            None => Some(0),
         }
     }
-}
 
-impl PartialEq for OwnedImportSource {
-    fn eq(&self, other: &Self) -> bool {
-        self.table == other.table && self.sid == other.sid
+    fn with_text(self, text: &'static str) -> Self {
+        Symbol::owned(text)
     }
-}
-
-impl Eq for OwnedImportSource {}
 
-impl ImportSource for OwnedImportSource {
-    fn table(&self) -> &str {
-        &self.table
+    fn with_symbol_id(self, _local_sid: SymbolId) -> Self {
```

Comment:
Do we really require this method now that it just places `Symbol::unknown_text()` for all symbols having unknown texts? 
Can we rather just have a new method called `with_unknown_text()`?

Reply:
Good question; `with_symbol_id` doesn't make much sense for `Symbol`, but it will make sense for `RawSymbolToken`, which represents a not-yet-resolved symbol from the input stream. The raw token might have a symbol ID or text.